src/test/java/com/microsoft/sqlserver/jdbc/connection/PoolingTest.java

```diff
@@ -139,4 +145,52 @@ public void testConnectionPoolClientConnectionId() throws SQLException {
 
         assertEquals(Id1, Id2, "ClientConnection Ids from pool are not the same.");
     }
+    
+    @Test
+    public void testHikariCP() throws SQLException {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(connectionString);
+        HikariDataSource ds = new HikariDataSource(config);
+
+        connect(ds);
```

Comment:
Bad choice of words on my part: this is actually a functional test. Remember when we changed a little bit of code to prevent creating lots of threads when many queries are executed rapidly? How do you know there is not another mistake like that still in the code? This test simply measures the amount of threads before and after executing many queries rapidly. Measuring is knowing.